### PR TITLE
feat: handle external url in `Link` on native

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
## Issue

NextJS's `Link` component works with external URLs. So, I think it makes sense to handle it on native as well.

### CODE REVIEW

**Comment by nandorojo** [comment]
My only question here is how we're detecting an external link. Is there a way to know how browsers do it? Is `startsWith('/')` correct?

**Comment by intergalacticspacehighway** [comment]
> if you accidentally start with artists/drake, what should we do?

yeah, I thought about it. Apparently, React Navigation requires links to start with `/`, or else it throws an error. I missed adding it in the description. Do you think it can still break?

https://github.com/react-navigation/react-navigation/blob/d7032ba8bb6ae24030a47f0724b61b561132fca6/packages/native/src/useLinkTo.tsx#L45

**Comment by nandorojo** [comment]
yeah but it will never error in this case because we don't forward the URL down. I'm wondering how the URL should be detected...

Should we check if it includes `://`? This would allow `http*://` as well as app deep links like `beatgig://`.

If we only match local requests for `/*`, then an empty `""` string would _not_ error from React Navigation (as it should) and will instead open a web browser on native. Meanwhile, an empty string would do nothing on Web.

There are edge cases, but just want to make sure we account for them properly to match expected behavior.

**Comment by intergalacticspacehighway** [comment]
Makes sense. I just took a look at how Next handles it. I think we could use the [isAbsoluteUrl](https://github.com/vercel/next.js/blob/77b5f79a4dff453abb62346bf75b14d859539b81/packages/next/shared/lib/router/router.ts#L143). It's doing a simple [regex test](https://github.com/vercel/next.js/blob/77b5f79a4dff453abb62346bf75b14d859539b81/packages/next/shared/lib/utils.ts#L314) to detect if URL is absolute which would make sense for external URLs. wdyt?

**Comment by nandorojo** [comment]
I think that makes sense!

**Comment by intergalacticspacehighway** [comment]
@nandorojo updated. Not sure why it's ignoring 2 deployments 🤔

File: src/link/core.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import type { ComponentProps, ComponentType } from 'react'
-import { Platform } from 'react-native'
+import { Platform, Linking } from 'react-native'
```
Comment (nandorojo) [inline src/link/core.tsx:3]: Last thing: could we tree-shake `openUrl` in a separate file such that it only imports `Linking` on native?

File: src/link/core.tsx [hunk 2]
```diff
@@ -65,12 +65,28 @@ function LinkCore({
...
       {children}
     </Component>
   )
 }
 
+// Scheme: https://tools.ietf.org/html/rfc3986#section-3.1
+// Absolute URL: https://tools.ietf.org/html/rfc3986#section-4.3
+const ABSOLUTE_URL_REGEX = /^[a-zA-Z][a-zA-Z\d+\-.]*?:/
+// Source - https://github.com/vercel/next.js/blob/77b5f79a4dff453abb62346bf75b14d859539b81/packages/next/shared/lib/utils.ts#L313
+export const isAbsoluteUrl = (url: string) => ABSOLUTE_URL_REGEX.test(url)
```
Comment (nandorojo) [inline src/link/core.tsx:90]: Don't think we need to export this.

**Comment by nandorojo** [comment]
Left some final minor comments then this looks good to me!

**Pull request merged**